### PR TITLE
feat(User-Agent): Fixes #40. Add --user-agent parameter which allows …

### LINKED ISSUE #40: [feature] User agent swap
Hey @droope 

Either there's no way to change my `User-agent` header, or I can't find it in the help docs. Any way you can add a functionality that allows me to change the `User-agent` of a scan?

Thanks!

### PR DESCRIPTION
**Pull request by kieran-github**
Hi, 

Here is a pull request that fixes #40 . 

Please have a review and let me know what you think.

### CODE REVIEW

**Comment by kieran-github** [comment]
Had some issues with the CI, i see there's a history of it failing for 3.3 - also fixed it - see: https://github.com/travis-ci/travis-ci/issues/9133.

**Comment by droope** [comment]
Hi

Thank you very much for your PR! This is the best PR I've received so far I think :)

I would like to request two changes prior to merging, if you could implement them I will certainly merge this code.

1. Python 3.3 is EOL. Could you, instead of performing those changes simply remove python 3.3 integration from travis? it may be as simple as removing that line.
2. I notice that you added the `user_agent` as a parameter for several functions, for example `enumerate_file_hash`. This parameter does not seem to be used, could you remove it from the kwargs variables and from the function definitions? There may be something I'm missing here, let me know if the variable is necessary! It's been a while since I've looked at this code. I'll leave you a note on all the functions I see this happening in.

I don't know why I can't see the travis integration on this PR, I'll have a look and see. 

Thanks!
Pedro

File: dscan/plugins/internal/base_plugin_internal.py [hunk 1]
```diff
@@ -711,7 +715,7 @@ def enumerate(self, url, base_url_supplied, scanning_method,
 
     def enumerate_plugins(self, url, base_url, scanning_method='forbidden',
             max_plugins=500, threads=10, verb='head', timeout=15,
-            hide_progressbar=False, imu=None, headers={}):
+            hide_progressbar=False, imu=None, headers={}, user_agent=''):
```
Comment (droope) [inline dscan/plugins/internal/base_plugin_internal.py:718]: Example of a function that does not use the user_agent param.

File: dscan/plugins/internal/base_plugin_internal.py [hunk 2]
```diff
@@ -722,7 +726,7 @@ def enumerate_plugins(self, url, base_url, scanning_method='forbidden',
 
     def enumerate_themes(self, url, base_url, scanning_method='forbidden',
             max_plugins=500, threads=10, verb='head', timeout=15,
-            hide_progressbar=False, imu=None, headers={}):
+            hide_progressbar=False, imu=None, headers={}, user_agent=''):
```
Comment (droope) [inline dscan/plugins/internal/base_plugin_internal.py:729]: Here as well

File: dscan/plugins/internal/base_plugin_internal.py [hunk 3]
```diff
@@ -732,7 +736,7 @@ def enumerate_themes(self, url, base_url, scanning_method='forbidden',
                 hide_progressbar, imu, headers)
 
     def enumerate_interesting(self, url, interesting_urls, threads=10,
-            verb='head', timeout=15, hide_progressbar=False, headers={}):
+            verb='head', timeout=15, hide_progressbar=False, headers={}, user_agent=''):
```
Comment (droope) [inline dscan/plugins/internal/base_plugin_internal.py:739]: Here as well

**Comment by kieran-github** [comment]
@droope See the updated changes. I removed it from all kwargs https://github.com/droope/droopescan/blob/master/dscan/plugins/internal/base_plugin_internal.py#L138. I had to remove two test cases as user_agent isn't part of call_args even though it is being used in https://github.com/droope/droopescan/blob/master/dscan/plugins/internal/base_plugin_internal.py#L237 (requests session initialization) . However it greatly simplifies the change and still passes all test cases - i think this is our best option.

**Comment by droope** [comment]
Thank you very much for this change, it looks good, and I know lots of people wanted it. I am going to merge it right in to the development branch.

It may be a little while before I am able to publish a new version, but I will do it as soon as I can.

Thanks!
Pedro

**Comment by droope** [comment]
I merged this into master! I noted your contribution on the changelog as well. https://github.com/droope/droopescan/blob/master/CHANGELOG

**Pull request merged**